R/archivr.R

```diff
@@ -284,66 +285,68 @@ view_archiv.fromText <- function (fp, method="wayback") {
 #'
 #' @param url The url to extract links from.
 #' @param method Either "wayback," "perma_cc" or "both".
+#' @param except A regular expression for URLs to exclude from extraction
 #' @export
 #' @return a dataframe containing the url, status, availability,
 #'   archived url(s) and timestamp(s)
 #' @examples
 #' # Wayback
-#' archiv.fromUrl("https://www-cs-faculty.stanford.edu/~knuth/retd.html")
+#' archivedURLs <- archiv.fromUrl("https://www-cs-faculty.stanford.edu/~knuth/retd.html", except="validator\\.w3\\.org")
 #'
 #' #perma.cc
 #' \dontrun{
 #' set_api_key("API KEY")
 #' set_folder_id("42")
-#' archiv.fromUrl("https://www-cs-faculty.stanford.edu/~knuth/retd.html", method="perma_cc")
+#' archivedURLs <- archiv.fromUrl("https://www-cs-faculty.stanford.edu/~knuth/retd.html", method="perma_cc")
 #' }
-archiv.fromUrl <- function (url, method="wayback") {
-  return(archiv(extract_urls_from_webpage(url), method))
+archiv.fromUrl <- function (url, method="wayback", except = NULL) {
+  return(archiv(extract_urls_from_webpage(url, except), method))
 }
 
 #' Save the links in a text file (docx, pdf, markdown) in perma.cc or Wayback.
 #'
 #' @param fp The filepath to extract links from.
 #' @param method Either "wayback," "perma_cc" or "both".
+#' @param except A regular expression for URLs to exclude from extraction
 #' @export
 #' @return a dataframe containing the url, status, availability,
 #'   archived url(s) and timestamp(s)
 #' @examples
 #' \dontrun{
 #' # Wayback
-#' archiv.fromText("testdoc.docx")
+#' archivedURLs <- archiv.fromText("testdoc.docx", except="doi\\.org\\/")
 #'
 #' #perma.cc
 #' set_api_key("API KEY")
 #' set_folder_id("42")
-#' archiv.fromText("testdoc.docx", method="perma_cc")
+#' archivedURLs <- archiv.fromText("testdoc.docx", method="perma_cc")
 #' }
-archiv.fromText <- function (fp, method="wayback") {
-  return(archiv(extract_urls_from_text(fp), method))
+archiv.fromText <- function (fp, method="wayback", except = NULL) {
+  return(archiv(extract_urls_from_text(fp, except), method))
 }
 
 #' Check whether a url is available in the Wayback Machine
 #'
 #' @param url The url to check.
 #' @importFrom jsonlite fromJSON
 #' @export
-#' @return a jsonlite object where
-#'   object$url is the original url.
-#'   object$$archived_snapshots$closest$status is the http status
-#'   object$archived_snapshots$closest$available is TRUE
-#'   object$archived_snapshots$closest$url is the archived url.
-#'   object$archived_snapshots$closest$timestamp is the last time the url
-#'     was crawled.
+#' @return a list containing
+#'   the original url.
+#'   the http status
+#'   TRUE if successful or FALSE
+#'   the archived url.
+#'   the last time the url was crawled.
 #' @examples
-#' from_wayback("https://www-cs-faculty.stanford.edu/~knuth/retd.html")
+#' checkStatus <- from_wayback("https://www-cs-faculty.stanford.edu/~knuth/retd.html")
```

Comment:
Same.

Reply:
Will fix, thanks, didn't see these comments before the latest commit.